openmrs/omod/src/main/java/org/openmrs/projectbuendia/servlet/DataExportServlet.java

```diff
@@ -159,86 +214,25 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
-                                } else {
-                                    s = Boolean.toString(value);
-                                }
-                                values[valueColumn] = s;
-                                values[valueColumn + 1] = s;
-                                values[valueColumn + 2] = s;
-                                return null;
-                            }
-
-                            @Override public Void visitText(String value) {
-                                if (value == null) {
-                                    value = "";
-                                }
-                                values[valueColumn] = value;
-                                values[valueColumn + 1] = value;
-                                values[valueColumn + 2] = value;
-                                return null;
-                            }
-
-                            @Override public Void visitDate(Date d) {
-                                String value;
-                                if (d == null) {
-                                    value = "";
-                                } else {
-                                    value = Utils.YYYYMMDD_UTC_FORMAT.format(d);
-                                }
-                                values[valueColumn] = value;
-                                values[valueColumn + 1] = value;
-                                values[valueColumn + 2] = value;
-                                return null;
-                            }
+                        int valueColumn = FIXED_HEADERS.length + index*COLUMNS_PER_OBS;
 
-                            @Override public Void visitDateTime(Date d) {
-                                String value;
-                                if (d == null) {
-                                    value = "";
-                                } else {
-                                    value = Utils.SPREADSHEET_FORMAT.format(d);
-                                }
+                        if (obs.getValueCoded() != null){
+                            Concept value = obs.getValueCoded();
+                            values[valueColumn] = NAMER.getClientName(value);
+                            values[valueColumn + 1] = value.getId();
+                            values[valueColumn + 2] = value.getUuid();
+                        } else {
+                            String value = (String) VisitObsValue.visit(obs, stringVisitor);
+                            if ((value != null) && (!value.isEmpty())) {
                                 values[valueColumn] = value;
                                 values[valueColumn + 1] = value;
                                 values[valueColumn + 2] = value;
-                                return null;
                             }
-                        });
+                        }
+                    }
+                    if (i == (encounters.size() - 1)){
```

Comment:
You can move the final `printer.printRecord` outside the loop and remove this conditional.